testutil/ciao-down/cloudinit.go

```diff
@@ -0,0 +1,427 @@
+
+ - curl -X PUT -d "Auto removing unused components" 10.0.2.2:{{.HTTPServerPort}}
+ - {{template "PROXIES" .}}apt-get auto-remove -y
+ - curl -X PUT -d "Unused components removed" 10.0.2.2:{{.HTTPServerPort}}
+
+ - curl -X PUT -d "Building ciao" 10.0.2.2:{{.HTTPServerPort}}
+ - sudo -u {{.User}} {{template "PROXIES" .}} GOPATH={{.GoPath}} /usr/local/go/bin/go get github.com/01org/ciao/...
+ - curl -X PUT -d "ciao built" 10.0.2.2:{{.HTTPServerPort}}
+
+ - curl -X PUT -d "Installing Go development utils" 10.0.2.2:{{.HTTPServerPort}}
+ - sudo -u {{.User}} {{template "PROXIES" .}} GOPATH={{.GoPath}} /usr/local/go/bin/go get github.com/fzipp/gocyclo github.com/gordonklaus/ineffassign github.com/golang/lint/golint github.com/client9/misspell/cmd/misspell
+ - curl -X PUT -d "Go development utils installed" 10.0.2.2:{{.HTTPServerPort}}
+
+ - chown {{.User}}:{{.User}} -R {{.GoPath}}
+
+ - curl -X PUT -d "Pulling ceph/demo" 10.0.2.2:{{.HTTPServerPort}}
+ - {{template "PROXIES" .}} docker pull ceph/demo
+ - curl -X PUT -d "Retrieved ceph/demo" 10.0.2.2:{{.HTTPServerPort}}
+
+ - mkdir -p /home/{{.User}}/local
+
+ - curl -X PUT -d "Downloading Fedora-Cloud-Base-24-1.2.x86_64.qcow2" 10.0.2.2:{{.HTTPServerPort}}
+ - {{template "PROXIES" .}}wget https://download.fedoraproject.org/pub/fedora/linux/releases/24/CloudImages/x86_64/images/Fedora-Cloud-Base-24-1.2.x86_64.qcow2 -O /home/{{.User}}/local/Fedora-Cloud-Base-24-1.2.x86_64.qcow2
+ - curl -X PUT -d "Fedora-Cloud-Base-24-1.2.x86_64.qcow2 Downloaded" 10.0.2.2:{{.HTTPServerPort}}
+
+ - curl -X PUT -d "Downloading CNCI image" 10.0.2.2:{{.HTTPServerPort}}
+ - {{template "PROXIES" .}}wget https://download.clearlinux.org/demos/ciao/clear-8260-ciao-networking.img.xz -O /home/{{.User}}/local/clear-8260-ciao-networking.img.xz
+ - curl -X PUT -d "CNCI image downloaded" 10.0.2.2:{{.HTTPServerPort}}
+
+ - curl -X PUT -d "Downloading latest clear cloud image" 10.0.2.2:{{.HTTPServerPort}}
+ - LATEST=$({{template "PROXIES" .}} curl -s https://download.clearlinux.org/latest) &&  {{template "PROXIES" .}} wget https://download.clearlinux.org/releases/"$LATEST"/clear/clear-"$LATEST"-cloud.img.xz -O /home/{{.User}}/local/clear-"$LATEST"-cloud.img.xz
+ - curl -X PUT -d "Latest clear cloud image downloaded" 10.0.2.2:{{.HTTPServerPort}}
+
+ - cd /home/{{.User}}/local && xz -T0 --decompress *.xz
+
+ - chown {{.User}}:{{.User}} -R /home/{{.User}}/local
+{{if len .GitUserName}}
+ - curl -X PUT -d "Setting git user.name" 10.0.2.2:{{.HTTPServerPort}}
+ - sudo -u {{.User}} git config --global user.name "{{.GitUserName}}"
+{{end}}
+{{if len .GitEmail}}
+ - curl -X PUT -d "Setting git user.email" 10.0.2.2:{{.HTTPServerPort}}
+ - sudo -u {{.User}} git config --global user.email {{.GitEmail}}
+{{end}}
+ - curl -X PUT -d "FINISHED" 10.0.2.2:{{.HTTPServerPort}}
+
+users:
+  - name: {{.User}}
+    gecos: CIAO Demo User
+    lock-passwd: true
+    shell: /bin/bash
+    passwd: $1$vzmNmLLD$04bivxcjdXRzZLUd.enRl1
```

Comment:
This hash needs to be removed.  It should never be used as lock-passwd is set to true.  I just left it in for debugging but now everything is working it can be deleted.